Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TimeOffRequestsController.cs

```diff
@@ -175,13 +168,17 @@ internal async Task ProcessTimeOffRequestsAsync(string isRequestFromLogicApp)
                                     }
                                 }
                             }
-                            while (hasMoreTimeOffs == true);
+                            while (hasMoreTimeOffs);
 
-                            if (timeOffRequestItems != null && timeOffRequestItems.Count > 0)
+                            if (timeOffRequestItems?.Count > 0)
                             {
+                                // get the team mappings for the team and pick the first because we need the Kronos Time Zone
+                                var teamMappings = await this.teamDepartmentMappingProvider.GetMappedTeamDetailsAsync(team).ConfigureAwait(false);
+                                var teamMapping = teamMappings.FirstOrDefault();
+
                                 foreach (var item in timeOffRequestItems)
                                 {
-                                    var timeOffReqStartDate = this.utility.UTCToKronosTimeZone(item.StartDateTime);
+                                    var timeOffReqStartDate = this.utility.UTCToKronosTimeZone(item.StartDateTime, teamMapping.KronosTimeZone);
```

Comment:
What happens if teamMapping is null?

Reply:
Same change applied as elsewhere and same issue with missing mappings.